src/anomalib/models/cfa/torch_model.py

```diff
@@ -78,6 +78,66 @@ def get_feature_extractor(backbone: str, return_nodes: list[str]) -> GraphModule
     return feature_extractor
 
 
+#Kmeans clustering algorithm implementation in PyTorch framework
+class KMeans_torch:
+    def __init__(self, n_clusters, max_iter=10):
+        """
+        Initializes the KMeans object.
+
+        Parameters:
+            n_clusters: The number of clusters to create.
+            max_iter: The maximum number of iterations to run the algorithm for.
+        """
+        self.n_clusters = n_clusters
+        self.max_iter = max_iter
+
+    def fit(self, X):
+        """
+        Runs the k-means algorithm on input data X.
+
+        Parameters:
+            X: A tensor of shape (N, D) containing the input data.
+            N is the number of data points 
+            D is the dimensionality of the data points.
+        """
+        N, D = X.shape
+
+        # Initialize centroids randomly from the data points
+        centroid_indices = torch.randint(0, N, (self.n_clusters,))
+        self.centroids = X[centroid_indices]
+
+        # Run the k-means algorithm for max_iter iterations
+        for i in range(self.max_iter):
+            # Compute the distance between each data point and each centroid
+            distances = torch.cdist(X, self.centroids)
+
+            # Assign each data point to the closest centroid
+            self.cluster_assignments = torch.argmin(distances, dim=1)
+
+            # Update the centroids to be the mean of the data points assigned to them
+            for j in range(self.n_clusters):
+                mask = self.cluster_assignments == j
+                if mask.any():
+                    self.centroids[j] = X[mask].mean(dim=0)
+                    
+        #thise line returns labels and centoids of the results, 
+        #alternative to Sklearn's cluster_centers_ & labels_ attributes            
+        return self.cluster_assignments, self.centroids
```

Comment:
Would it be possible to return `self`, similar to the `scikit-learn` implementation?